File: src/help.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "help.h"
#include "vars/colors.h"

int help() {
    printf(BOLD WHITE "               * -------- " BOLD LIGHT_CYAN"R" GREEN "e" YELLOW "c" BLUE "o" MAGENTA "n" CYAN "t" LIGHT_BLUE "r" RED "o" LIGHT_GREEN "l" LIGHT_MAGENTA "l" LIGHT_GREEN "e" YELLOW "d" RESET BOLD WHITE " -------- *\n\n");
 // printf(BOLD WHITE "                             * " BOLD LIGHT_CYAN "B" GREEN "E" YELLOW "" BLUE "T" MAGENTA"A" WHITE BOLD" *                         \n\n");
    printf(BOLD MAGENTA "             Usage: " WHITE "recontrolled" CYAN " [command] " RED "[<option>]      \n\n" RESET);
    printf(BOLD WHITE "            " BLUE "Basics                      " RED "Power-LED (pwr)      \n");
    printf(BOLD WHITE "* -------------------------- + -------------------------------- *\n");
    printf(BOLD WHITE "| " LIGHT_MAGENTA "help: Displays help" WHITE "        | " CYAN "ar: Activates Pwr-LED" WHITE "            |\n");
    printf(BOLD WHITE "| " LIGHT_MAGENTA "ver: Displays version" WHITE "      | " CYAN "dr: Deactivated Pwr-LED" WHITE "          |\n");
    printf(BOLD WHITE "| " LIGHT_MAGENTA "about: About Recontrolled" WHITE "  | " CYAN "br: Blinks Pwr-LED" WHITE "               |\n");
    printf(BOLD WHITE "| " LIGHT_MAGENTA "status: Shows status       " WHITE "| " CYAN "brf: Blinks Pwr-LED fast" WHITE "         |\n");
    printf(BOLD WHITE "| " LIGHT_MAGENTA "        of all leds      " WHITE "  | " CYAN "brs: Blinks Pwr-LED slow" WHITE "         |\n");
    printf(BOLD WHITE "| " LIGHT_MAGENTA "                         " WHITE "  | " CYAN "brff: Blinks Pwr-LED super fast" WHITE "  |\n");
    printf(BOLD WHITE "* -------------------------- + -------------------------------- *\n");
    printf(BOLD WHITE "                                                         \n");
    printf(BOLD WHITE "           " YELLOW "All LED's                   " LIGHT_GREEN "Activity-LED (act)    \n");
    printf(BOLD WHITE "* -------------------------- + -------------------------------- *\n");
    printf(BOLD WHITE "| " LIGHT_GREEN "lb: Blink LEDs syncronized" WHITE " | " MAGENTA "ag: Activates Act-LED" WHITE "            |\n");
    printf(BOLD WHITE "| " LIGHT_GREEN "lbs: Blink LEDs in switch" WHITE "  | " MAGENTA "dg: Deactivated Act-LED" WHITE "          |\n");
    printf(BOLD WHITE "| " LIGHT_GREEN "                         " WHITE "  | " MAGENTA "bg: Blinks Act-LED" WHITE "               |\n");
    printf(BOLD WHITE "| " LIGHT_GREEN "                         " WHITE "  | " MAGENTA "bgf: Blinks Act-LED fast" WHITE "         |\n");
    printf(BOLD WHITE "| " LIGHT_GREEN "                         " WHITE "  | " MAGENTA "bgs: Blinks Act-LED slow" WHITE "         |\n");
    printf(BOLD WHITE "| " LIGHT_GREEN "                         " WHITE "  | " MAGENTA "bgff: Blinks Act-LED super fast" WHITE "  |\n");
    printf(BOLD WHITE "* -------------------------- + -------------------------------- *\n");

    return 0;
}

File: src/basics/help.rs
use crate::constants::RECONTROLLED_STR;
use crate::constants::colors::{
    BLUE, BOLD, C_RESET, CYAN, DUMMY, LIGHT_GREEN, LIGHT_MAGENTA, PURPLE, RED, WHITE, YELLOW,
};

pub fn print_help() {
    println!(
        "                    {BOLD}{WHITE}* -------- {RECONTROLLED_STR} -------- *
                {BOLD}{PURPLE}Usage:{WHITE} recontrolled {CYAN}[command] {RED}[<options>]{C_RESET}

               {BOLD}{BLUE}Basics                           {RED}Power-LED (pwr){C_RESET}
{BOLD}{WHITE}* {DUMMY               }--------------------------------- + ----------------------------------- *
{BOLD}{WHITE}| {LIGHT_MAGENTA}help: Displays help             {WHITE}  | {CYAN}pwr on: Activates Pwr-LED          {WHITE} |
{BOLD}{WHITE}| {LIGHT_MAGENTA}version/ver: Displays version   {WHITE}  | {CYAN}pwr off: Deactivates Pwr-LED       {WHITE} |
{BOLD}{WHITE}| {LIGHT_MAGENTA}about: About Recontrolled       {WHITE}  | {CYAN}pwr blink: Blinks Pwr-LED          {WHITE} |
{BOLD}{WHITE}| {LIGHT_MAGENTA}status: Shows status of all LEDs{WHITE}  | {CYAN}pwr blink_fast: Fast blink         {WHITE} |
{BOLD}{WHITE}| {LIGHT_MAGENTA}                                {WHITE}  | {CYAN}pwr blink_slow: Slow blink         {WHITE} |
{BOLD}{WHITE}| {LIGHT_MAGENTA}                                {WHITE}  | {CYAN}pwr blink_super_fast: Super fast   {WHITE} |
{BOLD}{WHITE}| {LIGHT_MAGENTA}                                {WHITE}  | {CYAN}pwr blink_custom <ms>: Custom delay{WHITE} |
{BOLD}{WHITE}* {DUMMY               }--------------------------------- + ----------------------------------- *

              {BOLD}{YELLOW}All LEDs                         {LIGHT_GREEN}Activity-LED (act){C_RESET}
{BOLD}{WHITE}* {DUMMY             }--------------------------------- + ----------------------------------- *
{BOLD}{WHITE}| {LIGHT_GREEN}blink_sync: Blink LEDs sync      {WHITE} | {PURPLE}act on: Activates Act-LED          {WHITE} |
{BOLD}{WHITE}| {LIGHT_GREEN}blink_switch: Blink LEDs switch  {WHITE} | {PURPLE}act off: Deactivates Act-LED       {WHITE} |
{BOLD}{WHITE}| {LIGHT_GREEN}blink_sync_custom <ms>: Sync ... {WHITE} | {PURPLE}act blink: Blinks Act-LED          {WHITE} |
{BOLD}{WHITE}| {LIGHT_GREEN}  with custom delay              {WHITE} | {PURPLE}act blink_fast: Fast blink         {WHITE} |
{BOLD}{WHITE}| {LIGHT_GREEN}blink_switch_custom <ms>: ...    {WHITE} | {PURPLE}act blink_slow: Slow blink         {WHITE} |
{BOLD}{WHITE}| {LIGHT_GREEN}  Switch with custom delay       {WHITE} | {PURPLE}act blink_super_fast: Super fast   {WHITE} |
{BOLD}{WHITE}| {LIGHT_GREEN}                                 {WHITE} | {PURPLE}act blink_custom <ms>: Custom delay{WHITE} |
{BOLD}{WHITE}* {DUMMY             }--------------------------------- + ----------------------------------- {C_RESET}*
        "
    );
}
